spec/requests/api/v2/providers/courses/update_application_open_from.rb

```diff
@@ -0,0 +1,114 @@
+require "rails_helper"
+
+describe 'PATCH /providers/:provider_code/courses/:course_code' do
+  let(:jsonapi_renderer) { JSONAPI::Serializable::Renderer.new }
+
+  def perform_request(updated_applications_open_from)
+    jsonapi_data = jsonapi_renderer.render(
+      course,
+      class: {
+        Course: API::V2::SerializableCourse
+      }
+    )
+
+    jsonapi_data[:data][:attributes] = updated_applications_open_from
+
+    patch "/api/v2/providers/#{course.provider.provider_code}" \
+            "/courses/#{course.course_code}",
+          headers: { 'HTTP_AUTHORIZATION' => credentials },
+          params: {
+            _jsonapi: jsonapi_data
+          }
+  end
+  let(:organisation)      { create :organisation }
+  let(:provider)          { build :provider, organisations: [organisation] }
+  let(:user)              { create :user, organisations: [organisation] }
+  let(:payload)           { { email: user.email } }
+  let(:token)             { build_jwt :apiv2, payload: payload }
+
+  let(:course)            {
+    create :course,
+           provider: provider,
+           applications_open_from: applications_open_from
+  }
+
+  let(:applications_open_from) { DateTime.new(provider.recruitment_cycle.year.to_i, 1, 15).utc }
+
+  let(:credentials) do
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+
+  let(:permitted_params) do
+    %i[updated_applications_open_from]
+  end
+
+  before do
+    perform_request(updated_applications_open_from)
+  end
+
+  context "course has an updated applications_open_from" do
+    let(:updated_applications_open_from) { { applications_open_from: DateTime.new(provider.recruitment_cycle.year.to_i, 1, 1).utc } }
```

Comment:
Suggestion: set updated_applications_open_from to `applications_open_from` + some days/months

    let(:updated_applications_open_from) { { applications_open_from: applications_open_from + 5.days } }

This would make it clearer that it's different

Reply:
Done 👍